test/spec/authn/mfa-challenge.js

```diff
@@ -10,14 +10,40 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
+/* global document */
 
-jest.mock('../../../lib/util/misc', () => {
+jest.mock('lib/util', () => {
+  const actual = jest.requireActual('../../../lib/util');
   return {
+    ...actual,
     delay: () => { return Promise.resolve(); }
   };
 });
+
+jest.mock('lib/http', () => {
+  const actual = jest.requireActual('../../../lib/http');
+  return {
+    ...actual,
+    post: actual.post
+  };
+});
```

Comment:
Is this mock statement actually mocking anything out? It looks like we're just returning the actual implementation and then overwriting `post` with the actual implementation again, which sounds like a no-op.

Reply:
It is not mocking anything instanceously, but enables the ability to spy on the `post` method below

```js
context.httpSpy = jest.spyOn(mocked.http, 'post')
```